dds/DCPS/RTPS/Spdp.h

```diff
@@ -347,13 +347,53 @@ class OpenDDS_Rtps_Export Spdp : public DCPS::LocalParticipant<Sedp>
 #ifdef OPENDDS_SECURITY
   class SendStun : public DCPS::JobQueue::Job {
   public:
-    SendStun(SpdpTransport* tport, const ACE_INET_Addr& address, const STUN::Message& message) : tport_(tport), address_(address), message_(message) {}
+    SendStun(DCPS::RcHandle<SpdpTransport> tport,
+             const ACE_INET_Addr& address,
+             const STUN::Message& message)
+      : tport_(tport)
+      , address_(address)
+      , message_(message)
+    {}
     void execute();
   private:
-    SpdpTransport* tport_;
+    DCPS::RcHandle<SpdpTransport> tport_;
     ACE_INET_Addr address_;
     STUN::Message message_;
   };
+
+#ifndef DDS_HAS_MINIMUM_BIT
+  class IceConnect : public DCPS::JobQueue::Job {
```

Comment:
This could follow the pattern of `ChangeMulticastGroup` and not have distinct classes for Connect/Disconnect.